.github/workflows/pre-merge-integration-test.yml

```diff
@@ -0,0 +1,64 @@
+name: Pre-merge integration test
+on:
+  workflow_run:
+    workflows: ["Pre-merge checks"]
+    types:
+      - completed
+
+jobs:
+  deploy:
+    name: pre-merge-integration-tests
+    runs-on: ubuntu-latest
+    environment: di-ipv-cri-dev
+    timeout-minutes: 15
+    env:
+      AWS_REGION: eu-west-2
+      ENVIRONMENT: dev
+      STACK_NAME: address-cri-api
+      ROLLBACK_ACTION: "--disable-rollback"
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Set up JDK 11
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup SAM
+        uses: aws-actions/setup-sam@v1
+
+      - name: Assume temporary AWS role
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.GH_ACTIONS_ROLE_ARN }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Gradle build
+        run: ./gradlew build -x test -x spotlessCheck
+
+      - name: SAM build
+        run: sam build -t deploy/template.yaml
+
+      - name: Set short SHA
+        id: vars
+        run: echo "::set-output name=sha_short::$(git rev-parse --short HEAD)"
+
+      - name: SAM deploy integration test stack
+        run: |
+          STACK_NAME=${{ env.STACK_NAME }}-${{ steps.vars.outputs.sha_short }}
+          sam deploy -t deploy/template.yaml \
+            --no-fail-on-empty-changeset --no-confirm-changeset ${{ env.ROLLBACK_ACTION }} \
+            --signing-profiles SessionFunction=${{ secrets.SIGNING_PROFILE_NAME }} \
```

Comment:
Only post-merge code should be signed as the point about signing is that it proves that the application was built by the pipeline from reviewed code. If you apply the latest version of the pipeline stack it will prevent you from using the associated signing key in pre-merge scenarios: see PLAT-62

Reply:
👍 will remove for the dev environment